behave/features/steps/fala_end_to_end_tests.py

```diff
@@ -88,4 +102,42 @@ def step_impl_update_result_page(context, location, filter_label):
 def step_impl_error_shown_on_page(context):
     alert = context.helperfunc.find_by_css_selector(".alert-message")
     assert alert is not None
-    assert alert.text, "No results"
+    assert (
+        alert.text == "No results\nThere are no results matching your search criteria."
+    )
+
+
+@step(
+    'I am taken to the page corresponding to the "{location}" "{organisation}" search result'
+)
+def step_impl_result_page_with_multi_params(context, location, organisation):
+    result_number_paragraph = context.helperfunc.find_by_xpath(
+        '//p[@class="govuk-body"]'
+    )
+    current_url = context.helperfunc.driver().current_url
+    title_xpath = context.helperfunc.find_by_xpath(
+        "//html/body/div/main/div/div/h1"
+    ).text.replace("\n", " ")
+    result_container_xpath = context.helperfunc.find_by_xpath(
+        '//div[@class="search-results-container"]'
+    )
+    organisation = organisation.replace(" ", "+")
+    location = location.replace(" ", "+")
+
+    assert (
+        current_url
+        == f"""{CLA_FALA_URL}/?postcode={location}&name={organisation}&search="""
+    )
+    assert title_xpath == f"{FALA_HEADER}"
+    assert result_container_xpath, result_number_paragraph is not None
+
+
+@step('"{count}" result is visible on the results page')
+def step_impl_count_results_visible_on_results_page(context, count):
+    count = int(count)
+    listitems = context.helperfunc.find_many_by_xpath(
+        "//main/div/div[3]/div[1]/section/div/div[2]/div/ul/li"
```

Comment:
This targets a single list item in an under ordered list `<ul>` and not a list of list items `<li>`.

You can do find_many_by_xpath('//ul[@class="org-list"]/li`) which gets a list. Then perform a for loop through the list to count how many list items you have on a page. 

https://stackoverflow.com/questions/52092180/select-li-elements-from-ul-with-xpath Example of someone using a for loop on list items.

Reply:
I've tested this solution and it definitely returns a (python) list of <li> elements, which is why I just use the len() function to count the number of <li> elements. (Happy to demo) I've replace the line with this though '//ul[@Class="org-list"]/li` as it is a far neater and precise solution.